src/Features/CSharp/Portable/UsePatternMatching/CSharpIsAndCastCheckWithoutNameCodeFixProvider.cs

```diff
@@ -55,8 +55,10 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
                 getInnermostNodeForTie: true, cancellationToken: cancellationToken);
 
             var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var expressionTypeOpt = semanticModel.Compilation.GetTypeByMetadataName("System.Linq.Expressions.Expression`1");
+
             var (matches, localName) = CSharpIsAndCastCheckWithoutNameDiagnosticAnalyzer.Instance.AnalyzeExpression(
-                semanticModel, isExpression, cancellationToken);
+                semanticModel, isExpression, expressionTypeOpt, cancellationToken);
```

Comment:
Do we really need to pass `expressionTypeOpt` here?

If this was in an expression tree, then the codefix won't be triggered because no diagnostic would be reported. So unless I'm missing something, passing `null` would be okay.

Reply:
IMO, it's not great to have the analyzer and fixer exercise different codepaths.  It might lead to unexpected differences.  We should try to keep them the same. 

#resolved